person.rb

```diff
@@ -1,4 +1,11 @@
-class Person
+# rubocop:disable all
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
```

Comment:
- [ ] I got this error when I tried to test your decoration.

  ![image](https://user-images.githubusercontent.com/54848363/210575241-3833ac44-4256-44db-be5a-98ecb78c2fb5.png)

  You can fix this by adding `super()` to the initialize method in the Person class and creating the missing `correct_name` method:

  ```
  class Person < Nameable
  
    def initialize(age, name = 'Unknown', parent_permission: true)
      ...
      super()
    end

    def correct_name
      @name
    end
  ```

Reply:
I did as you suggested by adding the super method  and also the correct_name method thanks. 😄